syntax = "proto3";

service Payment {
    rpc CreatePayment (PaymentRequest) returns (PaymentReply) {}
}

// I might need to care about valua - I am not sure yet.
message PaymentRequest {
    float amount = 1; // negative values can be used for chargebacks or refunds
}

message PaymentReply {
    bool success = 1;
    optional string error = 2;
}